### PR TITLE
Fixing azure script

### PR DESCRIPTION
**Pull request by pawel-lmcb**

### CODE REVIEW

File: azure/cloud-shell/azure-cloud-shell-cloudadmin.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ else
 fi
 
 echo ". App registration id: $APP_ID"
-
+#EA appears to throw on the below...
```
Comment (pawel-lmcb) [inline azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:77]: Command group 'config' is experimental and under development. Reference and support levels: https://aka.ms/CLI_refstatus

Available subscriptions:

ID                                    NAME

--                                    ----

149e7793-7a00-4f52-9532-420354c39e68  2015 Azure EA

 

Enter subscription id to use with CloudAdmin: 149e7793-7a00-4f52-9532-420354c39e68

 

== Setup Starting..

 

== Setting up app 'CloudAdmin.Service.AUTOMATION'

 

== Checking if app exists

. Creating app registration

. App registration id: 87fa3a28-2cea-4b44-869a-3e476a3c0d13

ERROR: Resource '7b9e58d6-1337-4c4b-a34e-99d2b3d1b8e5' does not exist or one of its queried reference-property objects are not present.

== Checking app Service Principal

. Creating new SP

. Waiting for task completion

. Retrieving Exposed Permissions from SP

ERROR: Resource '755a0bd0-6700-40b7-bddc-e76866646393' does not exist or one of its queried reference-property objects are not present.

== Checking permissions of the app

ERROR: Resource '7b9e58d6-1337-4c4b-a34e-99d2b3d1b8e5' does not exist or one of its queried reference-property objects are not present.

charlie@Azure:~$

File: azure/cloud-shell/azure-cloud-shell-cloudadmin.sh [hunk 2]
```diff
@@ -132,6 +132,7 @@ if [[ "$ROLE_DEFINITIONS" != '[]' ]]; then
   az role definition update --role-definition "$UPDATED_ROLE" -o none
 else
   echo '. Creating custom role for CloudAdmin'
```
Comment (pawel-lmcb) [inline azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:134]: I added a new subscription PAY AS YOU GO and it's throwing:

pawel@Azure:~$ /bin/bash -c "$(curl -fsSL https://cloudadmin-onboarding.s3-us-west-2.amazonaws.com/azure/cloudshell/azure-cloud-shell-cloudadmin.sh)"
Command group 'config' is experimental and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Available subscriptions:
ID                                    NAME
--                                    ----
e2c82e54-295d-431a-bab6-33b808273d6b  Pay-As-You-Go linked to fb04
8a94d06a-9d4a-4a43-807c-f2050505fb04  Microsoft Azure Sponsorship 25k
815f7ae2-32ec-4659-a99f-6148901c79b8  Pay-As-You-Go linked to d96f
f648588f-78bf-482c-907c-f2f17b14d96f  Microsoft Azure Sponsorship 120k
98188db6-9b65-49be-a8d1-302779db1c90  Pay-As-You-Go linked to d96f too
b99eed74-d496-497f-9c15-95c68b84cf6a  PAYING FOR THIS USE FOR TESTING ONLY
d334b8d1-408a-4597-b67a-772b9ddbfce4  Pay-As-You-Go - May 20, 2021

Enter subscription id to use with CloudAdmin: d334b8d1-408a-4597-b67a-772b9ddbfce4

== Setup Starting..

== Setting up app 'CloudAdmin.Service.AUTOMATION'

== Checking if app exists
. Using existent app registration
. App registration id: 6afc187c-6c56-499c-9eb5-e9ef0b1eedf6
== Checking app Service Principal
. Using existent SP
. Retrieving Exposed Permissions from SP
== Checking permissions of the app
. Permissions already granted
== Assigning roles to the app
. Assigning Reservation Purchaser role to app
. Checking if custom role exists
. Creating custom role for CloudAdmin
A role definition cannot be updated with a name that already exists.

Comment (pawel-lmcb) [reply azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:134]: I think this is due to the fact that the LIST might only be subscription level, but role creation is global.

Comment (pawel-lmcb) [reply azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:134]: https://stackoverflow.com/questions/46241208/error-while-creating-new-role-assignment-in-azure

File: azure/cloud-shell/azure-cloud-shell-cloudadmin.sh [hunk 3]
```diff
@@ -252,6 +253,7 @@ if [[ "$EXPORT" != '[]' ]]; then
   EXPORT=$(az costmanagement export update --storage-account-id "$STORAGE_ACCOUNT_ID" --name "$EXPORT_NAME" --scope "$SUBSCRIPTION_SCOPE" --timeframe BillingMonthToDate --storage-container billing --recurrence Daily --schedule-status Active --recurrence-period from="$start_date" to="$end_date")
 else
   echo '. Creating new billing export'
+  #ECHO need to register resources here otherwise we get an error
```
Comment (pawel-lmcb) [inline azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:256]: SO, of course, when you do shit in the GUI it works, but apparently, if you do CLI you need to execute a command to actually register the resources lol
https://docs.microsoft.com/en-us/rest/api/cost-management/exports/create-or-update#exportdeliverydestination

Basically, you just need to register Microsoft.CostManagementExports with the subscription in question.

Comment (rcampos87) [reply azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:278]: added registering actions

File: azure/cloud-shell/azure-cloud-shell-cloudadmin.sh (hunk 2 above)
Comment (rcampos87) [reply azure/cloud-shell/azure-cloud-shell-cloudadmin.sh:146]: solved -- now we look for the role in all the subscriptions possible, and add a new scope to it

**Comment by pawel-lmcb** [comment]
@rcampos87 I opened the PR so can't approve in the UI, but it's approved!

**Review by rcampos87 — APPROVED** [review]

**Pull request merged**